api/data_explorer/controllers/export_url_controller.py

```diff
@@ -271,61 +271,93 @@ def _get_filter_query(filters):
-            field_type = current_app.config['EXTRA_FACET_INFO'][es_field_name][
+        if facet_id in current_app.config['FACET_INFO']:
+            field_type = current_app.config['FACET_INFO'][facet_id]['type']
+        elif facet_id in current_app.config['EXTRA_FACET_INFO']:
+            field_type = current_app.config['EXTRA_FACET_INFO'][facet_id][
                 'type']
         table_name, column, clause = _get_table_and_clause(
-            es_field_name, field_type, value, sample_file_column_fields)
-        if table_name in table_columns:
-            if column in table_columns[table_name]:
-                table_columns[table_name][column].append(clause)
-            else:
-                table_columns[table_name][column] = [clause]
+            facet_id, field_type, value, sample_file_column_fields)
+
+        if facet_id not in facet_table_clauses:
+            facet_table_clauses[facet_id] = {}
+        if table_name not in facet_table_clauses[facet_id]:
+            facet_table_clauses[facet_id][table_name] = []
+        facet_table_clauses[facet_id][table_name].append(clause)
+
+    table_wheres = {}
+    table_num = 1
+    table_select = '(SELECT %s FROM `%s` WHERE %s)'
+    query = 'SELECT DISTINCT t1.%s FROM ' % participant_id_column
+
+    def _append_to_query(existing, new, join, table_num):
+        return existing + join if table_num > 1 else existing + new
+
+    # Handle the clauses on a per-facet level.
+    for facet_id in facet_table_clauses.keys():
+        wheres = {}
+        for table_name in facet_table_clauses[facet_id].keys():
+            where = ''
+            for clause in facet_table_clauses[facet_id][table_name]:
+                if len(where) > 0:
+                    where += ' OR (%s)' % clause
+                else:
+                    where = '(%s)' % clause
+            wheres[table_name] = where
+
+        if len(wheres) == 1:
+            # All of the facet where clauses are for the same table, so add
+            # it to the table_wheres to be merged with other table-contained
+            # facets.
+            if table_name not in table_wheres:
+                table_wheres[table_name] = []
+            table_wheres[table_name].append(wheres[table_name])
         else:
-            table_columns[table_name] = {column: [clause]}
-
-    table_selects = []
-    participant_id_column = current_app.config['PARTICIPANT_ID_COLUMN']
-    for table_name, columns in table_columns.iteritems():
-        table_select = "(SELECT %s FROM `%s` WHERE %s)"
-        where_clause = ""
-        for column, clauses in columns.iteritems():
-            column_clause = ""
-            for clause in clauses:
-                if len(column_clause) > 0:
-                    column_clause = column_clause + " OR "
-                column_clause = column_clause + "(" + clause + ")"
+            # The facet where clauses span multiple tables. In order to OR
```

Comment:
```
# Samples Overview facet is different. Normally, different columns are AND'ed together.
# Different columns within Samples Overview facet are OR'ed together.
# OR is done using FULL JOIN in case columns are from different tables.
```